src/wp-includes/general-template.php

```diff
@@ -3413,6 +3413,121 @@ function wp_resource_hints() {
 	}
 }
 
+/**
+ * Prints resource preloads directives to browsers.
+ *
+ * Gives directive to browsers to preload specific resources that website will
+ * need very soon, this ensures that they are available earlier and are less
+ * likely to block the page's render.
+ *
+ * These performance improving indicators work by using `<link rel"preload">`.
+ *
+ * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload
+ * @link https://web.dev/preload-responsive-images/
+ *
+ * @since 6.0.0
+ */
+function wp_preload_links() {
+	/**
+	 * Filters domains and URLs for resource preloads.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array  $urls {
```

Comment:
@adamsilverstein I used $url the same way _wp_resource_hints_ uses it. Since both share a similar structure, I didn't want to introduce many changes.

Reply:
👍🏼 I see, poor naming choice but ok!

Reply:
> I used $url the same way [wp_resource_hints](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/general-template.php#L3336) uses it.

I see. However both places the name is no good, unfortunately. On top of that further down the var `$url` is reset/reused. Looks like a pretty poor choice :(

IMHO, best to make the names here descriptive, self-documenting and also fix the confusing naming in [wp_resource_hints](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/general-template.php#L3336).

Reply:
since we are renaming the function slightly based on your last comment to `wp_preload_resources`, thinking the variable name would be `$preload_resources`. How does that sound?